satpy/readers/seviri_l1b_icare.py

```diff
@@ -0,0 +1,276 @@
+        attr = self['/attr/Sub_Satellite_Longitude']
+        if isinstance(attr, np.ndarray):
+            attr = float(attr.astype(str))
+        return attr
+
+    @property
+    def projlon(self):
+        attr = self['/attr/Projection_Longitude']
+        if isinstance(attr, np.ndarray):
+            attr = float(attr.astype(str))
+        return attr
+
+    @property
+    def projection(self):
+        attr = self['/attr/Geographic_Projection']
+        if isinstance(attr, np.ndarray):
+            attr = str(attr.astype(str)).lower()
+        if attr != 'geos':
+            raise(NotImplementedError, "Only the GEOS projection is supported")
+        return attr
+
+    @property
+    def zone(self):
+        attr = self['/attr/Zone']
+        if isinstance(attr, np.ndarray):
+            attr = str(attr.astype(str)).lower()
+        return attr
+
+    @property
+    def res(self):
+        attr = self['/attr/Nadir_Pixel_Size']
+        if isinstance(attr, np.ndarray):
+            attr = str(attr.astype(str)).lower()
+        return float(attr)
+
+    @property
+    def endacq(self):
+        attr = self['/attr/End_Acquisition_Date']
+        if isinstance(attr, np.ndarray):
+            attr = str(attr.astype(str))
+        endacq = datetime.strptime(attr, "%Y-%m-%dT%H:%M:%SZ")
+        return endacq
+
+    @property
+    def stacq(self):
+        attr = self['/attr/Beginning_Acquisition_Date']
+        if isinstance(attr, np.ndarray):
+            attr = str(attr.astype(str))
+        stacq = datetime.strptime(attr, "%Y-%m-%dT%H:%M:%SZ")
+        return stacq
+
+    @property
+    def alt(self):
+        attr = self['/attr/Altitude']
+        if isinstance(attr, np.ndarray):
+            attr = attr.astype(str)
+        attr = float(attr)
+        # This is stored in km, convert to m
+        attr = attr * 1000.
+        return attr
+
+    @property
+    def geoloc(self):
+        attr = self['/attr/Geolocation']
+        if isinstance(attr, np.ndarray):
+            attr = attr.astype(str)
+        cfac = float(attr[0])
+        coff = float(attr[1])
+        lfac = float(attr[2])
+        loff = float(attr[3])
+        return [cfac, lfac, coff, loff]
+
+    def get_metadata(self, data, ds_info):
+        mda = {}
+        mda.update(data.attrs)
+        mda.update(ds_info)
+        geoloc = self.geoloc
+        mda.update({
+                    'start_time': self.stacq,
+                    'end_time': self.endacq,
+                    'platform_name': self.sensor_name[0],
+                    'sensor': self.sensor_name[1],
+                    'zone': self.zone,
+                    'satellite_altitude': self.alt,
+                    'cfac': geoloc[0],
+                    'lfac': geoloc[1],
+                    'coff': geoloc[2],
+                    'loff': geoloc[3],
+                    'resolution': self.res,
+                    'satellite_longitude': self.satlon,
+                    'projection_longitude': self.projlon,
```

Comment:
We have standardised the way to provide the different satellite longitudes here : https://satpy.readthedocs.io/en/latest/readers.html#metadata